File: benchmarks/vm/08_branch_loop.rb
t0 = Process.clock_gettime(Process::CLOCK_MONOTONIC)
total = 0
200_000.times do |i|
  r = i % 5
  if r == 0
    total += i
  elsif r == 1
    total -= i
  elsif r == 2
    total += i * 2
  else
    total += 1
  end
end
total_ms = (Process.clock_gettime(Process::CLOCK_MONOTONIC) - t0) * 1000
puts "total = #{total}"
puts format("BENCH_RESULT: %.3f ms", total_ms)

File: benchmarks/vm/08_branch_loop.py
import time

t0 = time.monotonic()
total = 0
for i in range(200000):
    r = i % 5
    if r == 0:
        total += i
    elif r == 1:
        total -= i
    elif r == 2:
        total += i * 2
    else:
        total += 1
total_ms = (time.monotonic() - t0) * 1000
print(f"total = {total}")
print(f"BENCH_RESULT: {total_ms:.3f} ms")
